### ISSUE TITLE
traefik-rtr uses a non-existent resolver: dns cloudflare

### ISSUE THREAD

**Issue by fogx** [htpcBeginner/docker-traefik#77]
hi,

i've been trying to follow your guide:
https://www.smarthomebeginner.com/traefik-2-docker-tutorial/

but keep running into issues with the dns resolver:

`level=error msg="the router traefik-rtr@docker uses a non-existent resolver: dns-cloudflare"`

i've tried recreating the acme.json, but that does not help.
Was there a config change, or why is this not working? Its pretty much straight out of the turorial...

**Comment by FA9US** [comment]
Hi Fgox,

I am busy with the same and battled my way through the errors. what worked for me was this

- Use the fulljuthub git hub file and mix with Synolog one to your liking
- create the proxy socket network suggested
- use the static IPs suggested in the comment of the code
- then take off the Traefik bit of the code and copy from the original guide
- Include the secret files for the cloudfare API and the other .env files
- chmod 600 is important on acme.json file
- Run the code and check acme.json to confirm the staging test ssl is in the file
- nano and clean the file again
- comment the Letsencrypt staging link in the YML and run again
- If all done well you have your cert but still failing services
- Copy the Traefik from git file here back again
- Uncomment the 8080 dashboard so you can see whats going on in traefik
- Make sure you got the traefik2\rules fixed

Lots of things in the full builds dont work and you just have to trial and error.
errrors i found:
- i noticed Sabnzbd doesnt connect with SSL
- MariaDB doesnt get the password. and boots passwordless
- radarr doesnt work
- authelia wont work without Oauth because its not independant tool
- docker-GC and mosquitto wont work and you need to manually fix them

my issues are more on the oauth and authelia.

**Comment by fogx** [comment]
hey @FA9US thanks for your detailed reply!
i've got the letsEncrypt certificate working now, but i can't access my whoami service and the dashboard only works because of api.insecure and isn't encrypted... so basically i have the certificate, but i can't use it for anything (eventhough i have the http-redirect running, which should reroute all traffic to https=SSL??!)

what do you mean by "got the traefik2/rules fixed"? i got the middlewares and middleware-chains files in there and i think they work (?)

<details>
<summary> New docker-compose</summary>
version: "3"

services:
  traefik:
    #https://www.smarthomebeginner.com/traefik-2-docker-tutorial/
    image: traefik:latest
    container_name: traefik
    command: # CLI arguments
      - --global.sendAnonymousUsage=false
      - --api=true
      - --api.insecure=true
      ## Entrypoints Settings - https://docs.traefik.io/routing/entrypoints/#configuration ##
      - --entrypoints.http.address=:80 # <== Defining an entrypoint for port :80
      - --entrypoints.https.address=:443 # <== Defining an entrypoint for https on port :443
      - --entryPoints.traefik.address=:8080
        # Allow these IPs to set the X-Forwarded-* headers - Cloudflare IPs: https://www.cloudflare.com/ips/
      # - --entrypoints.https.forwardedHeaders.trustedIPs=173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/12,172.64.0.0/13,131.0.72.0/22

      ## providers
      - --providers.docker=true
      - --providers.docker.exposedByDefault=false
      - --providers.docker.network=traefik_proxy # Defines a default docker network to use for connections to all containers.
      - --providers.file.directory=/rules # Load dynamic configuration from one or more .toml or .yml files in a directory.
      - --providers.file.watch=true # Only works on top level files in the rules folder

      ## Certificate Settings (Let's Encrypt) -  https://docs.traefik.io/https/acme/#configuration-examples ##
      # - --certificatesResolvers.dns-cloudflare.acme.caServer=https://acme-staging-v02.api.letsencrypt.org/directory # LetsEncrypt Staging Server - uncomment when testing
      - --certificatesResolvers.dns-cloudflare.acme.email=$CLOUDFLARE_EMAIL
      - --certificatesResolvers.dns-cloudflare.acme.storage=/acme.json
      - --certificatesResolvers.dns-cloudflare.acme.dnsChallenge.provider=cloudflare
      
      ## logging
      - --log=true
      - --log.level=DEBUG # (Default: error) DEBUG, INFO, WARN, ERROR, FATAL, PANIC #change to warn when done setting up
      - --accessLog=true
      - --accessLog.filePath=/traefik.log
      - --accessLog.bufferingSize=100 # Configuring a buffer of 100 lines
      - --accessLog.filters.statusCodes=400-499

    labels:
      - "traefik.enable=true"
      # HTTP-to-HTTPS Redirect
      - "traefik.http.routers.http-catchall.entrypoints=http"
      - "traefik.http.routers.http-catchall.rule=HostRegexp(`{host:.+}`)"
      - "traefik.http.routers.http-catchall.middlewares=redirect-to-https"
      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"
      # HTTP Routers
      - "traefik.http.routers.traefik-rtr.entrypoints=https"
      - "traefik.http.routers.traefik-rtr.rule=Host(`traefik.$DOMAINNAME`)"
      - "traefik.http.routers.traefik-rtr.tls=true"
      - "traefik.http.routers.traefik-rtr.tls.certresolver=dns-cloudflare" # Comment out this line after first run of traefik to force the use of wildcard certs
      - "traefik.http.routers.traefik-rtr.tls.domains[0].main=$DOMAINNAME"
      - "traefik.http.routers.traefik-rtr.tls.domains[0].sans=*.$DOMAINNAME"
      ## Services - API
      - "traefik.http.routers.traefik-rtr.service=api@internal"
      ## Middlewares
      - "traefik.http.routers.traefik-rtr.middlewares=chain-basic-auth@file"
    networks:
      - traefik_proxy
      - default
    security_opt:
      - no-new-privileges:true
    ports:
      - "80:80" # <== http
      - "443:443" # <== https
      - "8080:8080" # <== :8080 is where the dashboard runs on
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - $DOCKERDIR/traefik/rules:/rules
      - $DOCKERDIR/traefik/acme/acme.json:/acme.json
      - $DOCKERDIR/traefik/traefik.log:/traefik.log
      - $DOCKERDIR/shared:/shared
    environment:
      - CF_API_EMAIL=$CLOUDFLARE_EMAIL
      - CF_API_KEY=$CLOUDFLARE_API_KEY
    restart: unless-stopped

  whoami:
    # A container that exposes an API to show its IP address
    image: containous/whoami
    networks:
      - traefik_proxy
    labels:
      - traefik.enable=true
      ## HTTP Routers
      - "traefik.http.routers.whoami.entrypoints=http"
      - "traefik.http.routers.whoami.rule=HostHeader(`who.$DOMAINNAME`)"
      - traefik.http.routers.whoami.tls=true

networks:
  traefik_proxy:
    external:
      name: traefik_proxy
  default:
    driver: bridge

</details>
<details>
<summary> middlewares.toml</summary>
[http.middlewares]
  [http.middlewares.middlewares-basic-auth]
    [http.middlewares.middlewares-basic-auth.basicAuth]
      realm = "Traefik2 Basic Auth"
      usersFile = "/shared/.htpasswd" #be sure to mount the volume through docker-compose.yml

</details>

all the log says is:
` msg="No ACME certificate generation required for domains [\"xxx.yyy\" \"*.xxx.yyy\"]." providerName=dns-cloudflare.acme`
and accessing the pages returns: `ERR_CONNECTION_TIMED_OUT`

**Comment by FA9US** [comment]
yes thats what i meant with renaming the example files.  the log seems fine. if you check your cloudflare you now have lots of subdomains. the FW need to be open so you can go outbound 80 and 443 on your host. then when you go to one of the subdomains you can see the SSL being applied which is goed bringing you to where i am as we speak. both Oauth and Authelia are failing for me. 

to confirm your setup you can simply overwrite the YML where one of the below lables are used.
no auth besically is for traefik to directly relay without any middlewares. so this way you can test your SSL

i can do both no auth and basic auth but my Oauth and authelia fails.

oauth i get service doesnt exist
and authelia i get access denied when it relays to the container... 

chain-no-auth@file
chain-basic-auth@file
chain-oauth@file
chain-authelia@file


i like this so much ive spent i think too much on this (80+ hours 📦  to figure it out hahaha

sabnzbd ive got external now. did connecting to SSL work for you?

**Comment by parthp1808** [comment]
> the issue with the dns cloudflare is resolved though. You'll have to post more infos to help you! some log output or something.

How you got it fixed? I am having exact same issue right now and I am just trying to setup traefik right now (with staging cloudflare) to test the setup but can't get it working

**Comment by simonholl** [comment]
Also dealing with this.
This issue doesn't seem to actually affect any functionality as all my subdomains are accessible with correct certs and are automatically renewed when required.

How did you resolve it? Please let us know.

**Comment by fogx** [comment]
> Also dealing with this.
> This issue doesn't seem to actually affect any functionality as all my subdomains are accessible with correct certs and are automatically renewed when required.
> 
> How did you resolve it? Please let us know.


this is my current setting for cf


      ## Certificate Settings (Let's Encrypt) -  https://docs.traefik.io/https/acme/#configuration-examples ##      # - --certificatesResolvers.dns-cloudflare.acme.caServer=https://acme-staging-v02.api.letsencrypt.org/directory # LetsEncrypt Staging Server - uncomment when testing      - --certificatesResolvers.dns-cloudflare.acme.email=$CLOUDFLARE_EMAIL      - --certificatesResolvers.dns-cloudflare.acme.storage=/acme.json      - --certificatesresolvers.dns-cloudflare.acme.dnschallenge=true      - --certificatesResolvers.dns-cloudflare.acme.dnsChallenge.provider=cloudflare

**Comment by simonholl** [comment]
Thanks, I will have a look and report if I manage to resolve it.

**Comment by simonholl** [comment]
I haven't really resolved it, but I think I understand what is happening. Hopefully someone that knows better can confirm.

I am getting the 'non-existent resolver' message for all services except Traefik. 
Traefik is configured to use 'dns-cloudflare' as the default certresolver for all services instead of specifying individually under each service. 
Perhaps Traefik is still looking for a defined certresolver under each service (except itself) and marking it as non-existent?

Can I safely ignore this message then? All services are available and working via my domain.

**Comment by Genieplumb** [comment]
I've just got this finally up and working, had this error and many many more, what I've found is to go over the code entered that you've altered to suit your setup again and again. The error reporting is often not helpful and at times vague. Check your access to either the .env file (the root of the docker-compose folder?) or the secrets file. this was where my two evenings were spent sorting this error for me. Also when it's up don't assume it works on all browsers, or even check that it's not working on at least two browsers too. Best of the best luck in sorting. (this is the way)
